### ISSUE TITLE
Cannot pull Apotheosis Affix items out of a terminal

### ISSUE THREAD

**Issue by brisingraerowing** [CyclopsMC/IntegratedTerminals#106]
#### Issue type:

- :bug: Bug 

____

#### Short description:

Trying to put an Apotheosis Affix item (armor, tool, etc) from a terminal into my inventory doesn't do anything.

#### Steps to reproduce the problem:

1. Install ID, IT and Apotheosis
2. Find / Make some Affix Items
3. Put Affix Items in storage
4. Try to pull them out

#### Expected behaviour:

Items are extractable.

____

#### Versions:

- This mod: 1.4.8
- Minecraft: 1.19.2
- Forge: 43.2.14

#### Log file:
[Latest Log](https://gist.github.com/brisingraerowing/9ff74dccfcab440631052d62473f888e)

I didn't see anything of any interest.

**Comment by rubensworks** [comment]
Thanks for reporting!

**Comment by rubensworks** [comment]
Since this doesn't occur with any other items I've heard of, is there anything special about these items that would block them from being pulled?
Are they stored in a special chest or something?
Might be interesting to also report this issue to the Apotheosis dev; perhaps that person has some additional insights.

**Comment by brisingraerowing** [comment]
They have a ton of additional NBT data defining the Affixes. They were stored in Sophisticated Storage Netherite Barrels, as was everything else that could still be pulled out. I think the large amount of NBT may be at least part of the issue.

**Comment by rubensworks** [comment]
Hi @Shadows-of-Fire 👋,
Do you have any insights that could help solve this issue?

Do these affix items do anything in particular with NBT (such as evolving over time)?
Or can NBT tags just be very large? (if they are just large, it _should_ still work fine in terminals...)

**Comment by Shadows-of-Fire** [comment]
"A ton of additional NBT data" isn't even particularly accurate - affixes are stored as additional nbt similar to enchantments ( a list of pairs of names -> levels)

The NBT data does not change over time unless the item is manipulated in a reforging table or smithing table.

Items with gems socketed into them record in nbt the full item in their nbt, but even then it isn't particularly large, a backpack would easily have more NBT data

This is an export of a mythic item with 3 socketed gems, if it helps https://gist.github.com/Shadows-of-Fire/16bbd553a10b2bef40c2b58e76141942

It's from /data, so it isn't proper json, but it should illustrate the data volume

<blockquote><img src="https://github.githubassets.com/images/modules/gists/gist-og-image.png" width="48" align="right"><div><img src="https://github.githubassets.com/favicons/favicon.svg" height="14"> Gist</div><div><strong><a href="https://gist.github.com/Shadows-of-Fire/16bbd553a10b2bef40c2b58e76141942">item.json</a></strong></div><div>GitHub Gist: instantly share code, notes, and snippets.</div></blockquote>

**Comment by rubensworks** [comment]
Thanks for the input @Shadows-of-Fire!

I don't see anything out of the ordinary, so I'll have some debugging to do on my end.

Before I do that, @brisingraerowing, could you check if the error also occurs on your end with a pack containing just ID and Apotheosis, to rule out config/mod conflicts?

**Comment by brisingraerowing** [comment]
I can confirm with just ID / ITerm / ITunnel, Apotheosis, and JEI (and the other requisite library mods) any affix item cannot be pulled out of the terminal.

ETA: I also had Sophisticated Storage in there, though the issue also occurs with Vanilla chests.

**Comment by rubensworks** [comment]
Thanks for checking!

**Comment by flann** [comment]
FYI - I think this also affects ability totems in functional storage armories. 
But not always. I had ability totems in an armory with a bunch of other stuff in it. I was able to pull totems out of that armory via a storage terminal. However, when I pulled the totems out of that armory into another one and put a storage terminal on it, I was no longer able to pull the totems out of that armory.
I've seen this happen with apotheosis gems, too.
